### PR TITLE
fix static analysis warnings and stop using nulls everywhere

### PR DESCRIPTION
**Pull request by artbristol**
Got rid of some null-paranoia.

I am prepared for pushback on changes in `MessageProvider` but I think there is already too much 'the show must go on' mentality in the code.

### CODE REVIEW

File: k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java [hunk 1]
```diff
@@ -2379,7 +2379,7 @@ private void processDraftMessage(LocalMessage message) throws MessagingException
         // Decode the identity header when loading a draft.
         // See buildIdentityHeader(TextBody) for a detailed description of the composition of this blob.
         Map<IdentityField, String> k9identity = new HashMap<IdentityField, String>();
-        if (message.getHeader(K9.IDENTITY_HEADER) != null && message.getHeader(K9.IDENTITY_HEADER).length > 0 && message.getHeader(K9.IDENTITY_HEADER)[0] != null) {
+        if (message.getHeader(K9.IDENTITY_HEADER).length > 0 && message.getHeader(K9.IDENTITY_HEADER)[0] != null) {
```
Comment (cketti) [inline k9mail/src/main/java/com/fsck/k9/activity/MessageCompose.java]: Can you please extract `message.getHeader(K9.IDENTITY_HEADER)` to a local variable.

**Comment by cketti** [comment]
I believe crashes in `MessageProvider` kill the K-9 Mail process. If that's true we shouldn't allow third-party apps to crash our process using "malformed" queries to the content provider.

**Comment by artbristol** [comment]
@cketti fixed your comments

**Comment by cketti** [comment]
:+1:

**Pull request merged**